app/models/round.rb

```diff
@@ -26,31 +27,12 @@ def decrement_position
 
   private
 
-  def unique_round_for_player_in_game
-    if Round.where(game_player_id:, round_type:)
-            .where.not(id:)
-            .joins(:game_player)
-            .where(game_players: { game_id: game_player.game_id })
-            .any?
-      errors.add(:base, 'Round with the same (player_id, game_id, round_type) already exists!')
-    end
-  end
-
   def set_position
-    last_position = game.rounds.where.not(id: nil).maximum(:position) || 0
+    last_position = game_player.game.rounds.where.not(id: nil).maximum(:position) || 0
```

Comment:
What added the `gameplayer.` here? You already have direct relation to the game, via game_players

Reply:
I got an error last time, but it seems something else was the reason. I got game_player out and it works only with game.rounds.